platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/platformstatus/PlatformStatusStateMachine.java

```diff
@@ -102,25 +102,28 @@ public PlatformStatusStateMachine(
     private PlatformStatusLogic getNewLogic(@NonNull final PlatformStatusAction action) {
         Objects.requireNonNull(action);
 
+        final Class<? extends PlatformStatusAction> actionClass = action.getClass();
+
         try {
```

Comment:
Random thought, is it possible to rewrite this using a switch statement? Might be more aesthetically pleasing that way.